File: probando/src/probando/persona.java
/**
 *
 * @author elena
 */
package probando;

public class persona {
    private static int n_personas= 0;
    private String nombre;
    private int anios;
    static String planeta= "Tierra";
    
    public persona(String name, int anios){
        this.nombre= name;
        this.anios= anios;
        n_personas += 1;
    }
    
    static int get_n_personas(){
        return n_personas;
    }
    
    protected String get_nombre(){
        return nombre;
    }
    
    int get_anios(){
        return anios;
    }
 
    protected void set_nombre(String n){
        this.nombre= n;
    }
    
    static String getPlaneta(){
        return planeta;
    }
    
    Object hablar(){
        return "bla bla bla desde persona";
    }
}

File: Ejemplos_ruby/lib/persona.rb
module Ejemplos_ruby
  class Persona
    @@n_personas= 0
    attr_accessor :dni, :nombre

    def initialize(d, n)
      @dni= d
      @nombre= n
      @@n_personas += 1
    end

    def self.get_n_personas
      @@n_personas
    end

    def hablar(*interlocutores)
      if interlocutores != nil
        puts interlocutores.to_s
      end
      "bla bla bla"
    end

    protected :nombre
    private :dni
  end
end